config/config.go

```diff
@@ -549,6 +554,31 @@ func (c *GlobalConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return fmt.Errorf("%w for global config", err)
 	}
 
+	// If there is no configuration for global.gogc, use a fallback logic.
+	if gc.GoGC == 0 {
+		// By default, use the global default.
+		gc.GoGC = DefaultGlobalConfig.GoGC
+		goGCEnv := os.Getenv("GOGC")
+		// If the GOGC env var is set, use the same logic as upstream Go.
+		if goGCEnv != "" {
+			// Special case for GOGC=off.
+			if strings.ToLower(goGCEnv) == "off" {
+				gc.GoGC = -1
+			} else {
+				i, err := strconv.Atoi(goGCEnv)
+				if err == nil {
+					gc.GoGC = i
+				}
+			}
+		}
+	}
+	// Write the new setting out to the ENV var for runtime API output.
```

Comment:
How about setting this next to `oldGoGC := debug.SetGCPercent` for better visibility and to move it away from this `UnmarshalYAML`?

Reply:
Sure, that also works.